src/transformers/models/bart/modeling_tf_bart.py

```diff
@@ -963,7 +963,8 @@ def call(
             else self.config.output_hidden_states
         )
         return_dict = inputs["return_dict"] if inputs["return_dict"] is not None else self.config.return_dict
-        if not use_cache:
+
+        if not use_cache or past_key_values is None:
```

Comment:
This was actually not correct neither in PT nor in TF. If use_cache=True => this does not mean that decoder_input_ids.shape[-1] == 1 and that we rely on `past_key_value`. The very first time `decoder_input_ids` are passed with `use_cache=True`, we need need to use a causal mask if `decoder_input_ids' seq len is > 1 or the result would be incorrect.